clientd3d/server.c

```diff
@@ -359,7 +365,12 @@ void ExtractObjectNoLight(char **ptr, object_node *item)
    Extract(ptr, &item->icon_res, SIZE_ID);
    Extract(ptr, &item->name_res, SIZE_ID);
    Extract(ptr, &item->flags, 4); // includes drawfx_mask bits
-
+   Extract(ptr, &item->special_type, SIZE_ID);
```

Comment:
ExtractObjectNoLight is an abomination---it means code duplication, and requires you to make changes twice, as in this case.  We should make a separate change that merges the code in ExtractObject and ExtractObjectNoLight.  You can do it if you'd like, or I can do it, and then you'd have to merge my change into your branch (and probably resolve a merge conflict).

Reply:
Refactored `ExtractObjectNoLight` as requested so it's contained within this PR.